### PR TITLE
Cloudwatch logs role

### PR DESCRIPTION
**Pull request by alexduf**
This allow us to bake images with the cloudwatch-log agent.

It also provide a utility script to reduce the boiler plate in the user-data within the cloudformation.
It assumes the distribution is using systemd

The readme should explain it all, so if it isn't clear please let me know I'll do my best to improve it.

### CODE REVIEW

File: app/components/AppComponents.scala [hunk 1]
```diff
@@ -42,7 +42,7 @@ class AppComponents(context: Context)
     val awsCreds = new AWSCredentialsProviderChain(
       new ProfileCredentialsProvider("deployTools"),
       new ProfileCredentialsProvider(),
-      new InstanceProfileCredentialsProvider
+      InstanceProfileCredentialsProvider.getInstance()
```
Comment (philwills) [inline app/components/AppComponents.scala]: Why is this change required?

File: roles/cloud-watch-logs/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+This role will install the [cloudwatch](http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/AgentReference.html) agent.
+
+It also provide a utility script to configure your logs in `/opt/cloudwatch-logs/configure-logs`
+The arguments are:
+ - an id, to distinguish between log file (for instance `application` and `gc`)
+ - the stack
```
Comment (philwills) [inline roles/cloud-watch-logs/README.md]: We could default these from tags, but I'm not sure that's worth the extra complexity.

File: app/components/AppComponents.scala (hunk 1 above)
Comment (alexduf) [inline app/components/AppComponents.scala]: It actually isn't, but the direct creation of the `InstanceProfileCredentialsProvider` is deprecated now

File: roles/cloud-watch-logs/README.md (hunk 2 above)
Comment (alexduf) [inline roles/cloud-watch-logs/README.md]: I hadn't thought of it, but that would be a great improvement. I'd like to merge it as is though as I don't think I should spend more time on it.

File: app/components/AppComponents.scala (hunk 1 above)
Comment (philwills) [inline app/components/AppComponents.scala]: Perfect, thanks.
